std/range/package.d

```diff
@@ -4688,7 +4688,13 @@ if ((isIntegral!(CommonType!(B, E)) || isPointer!(CommonType!(B, E)))
         && isIntegral!S)
 in
 {
-    assert(!(step == 0 && begin != end));
+    import std.conv : unsigned;
+
+    assert(step != 0 || begin == end);
+    if (step > 0)
+        assert(unsigned((end - begin) / step) <= size_t.max);
+    else if (step < 0)
+        assert(unsigned((end - begin) / -step) <= size_t.max);
```

Comment:
No, this isn't right. unsigned math works fine if it's addition/subtraction, but does not work with multiplication/division. You need the if structure.